### DISCUSSION TITLE
Dynamically change a link/flow

### DISCUSSION THREAD

**Discussion by PFlorient** [clientIO/joint#2356] [Q&A] [answered]
Hello,

I'm contacting you because I'd like to set up a select list in which I can change the type of my flow. 

The problem is that I use the flow vsm provided by the library ("electronic information flow", "Information Flow", "Shipment" more info here: https://resources.jointjs.com/docs/rappid/v3.6/shapes.vsm.html) 
while checking the documentation and some demos, I saw that the flows come from BPMN2 (https://resources.jointjs.com/docs/rappid/v3.5/shapes.bpmn2.html). 

I noticed that they are one and the same flow where you only change the `flowtype`. 
Here I don't have the impression that this is the case and I can't see how it's possible to develop this. 

I'd imagined directly destroying the existing flow and replacing it with another, but that's probably overkill.

Thanks in advance

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
I suggest removing the existing link and creating a new one with the same `id`, `source` and `target`. Here's an [example](https://codepen.io/jointjs/pen/BaYOREQ) with elements (not links).

The `flowtype` attribute was implemented for the older version of [BPMN](https://resources.jointjs.com/docs/rappid/v3.7/shapes.bpmn.html#shapes.bpmn.flow). The same approach was not taken for the newer version of BPMN2 because the dynamic `flowtype` made it difficult to style.

The link internally listened for the `change:flowtype` event attribute and updated the its presentation attributes based on that type.
